### PR TITLE
Breaking: #96659 - Registration of cObjects via TYPO3_CONF_VARS

### PR DESCRIPTION
**Pull request by linawolf**
https://docs.typo3.org/c/typo3/cms-core/main/en-us/Changelog/12.0/Breaking-96659-RegistrationOfCObjectsViaTYPO3_CONF_VARS.html

refs https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/issues/1624

releases: main

### CODE REVIEW

**Review by susannemoog — APPROVED** [review]

File: Documentation/ApiOverview/GlobalValues/Typo3ConfVars/FE.rst [hunk 1]
```diff
@@ -767,15 +767,21 @@ $GLOBALS['TYPO3_CONF_VARS']['FE']['contentRenderingTemplates']
...
 ===================================================
 
-.. confval:: ContentObjects
+.. versionchanged:: 12.0
 
-   :Path: $GLOBALS['TYPO3_CONF_VARS']['FE']
-   :type: array
-   :Default: []
+   The global variable `$GLOBALS['TYPO3_CONF_VARS']['FE']['ContentObjects']` is
+   not interpreted anymore.
```
Comment (mbrodala) [inline Documentation/ApiOverview/GlobalValues/Typo3ConfVars/FE.rst:773]:
```suggestion
   not used anymore.
```

File: Documentation/ApiOverview/GlobalValues/Typo3ConfVars/FE.rst [hunk 2]
```diff
@@ -767,15 +767,21 @@ $GLOBALS['TYPO3_CONF_VARS']['FE']['contentRenderingTemplates']
...
+
+TypoScript ContentObjects (cObjects) like :typoscript:`TEXT` or
```
Comment (mbrodala) [inline Documentation/ApiOverview/GlobalValues/Typo3ConfVars/FE.rst:775]:
```suggestion
TypoScript content objects (`cObject`) like :typoscript:`TEXT` or
```

File: Documentation/ApiOverview/GlobalValues/Typo3ConfVars/FE.rst [hunk 3]
```diff
@@ -767,15 +767,21 @@ $GLOBALS['TYPO3_CONF_VARS']['FE']['contentRenderingTemplates']
...
+:typoscript:`HMENU` get registered via the service configuration:
```
Comment (mbrodala) [inline Documentation/ApiOverview/GlobalValues/Typo3ConfVars/FE.rst:776]:
```suggestion
:typoscript:`HMENU` are registered as services:
```

File: Documentation/ApiOverview/GlobalValues/Typo3ConfVars/FE.rst [hunk 4]
```diff
@@ -767,15 +767,21 @@ $GLOBALS['TYPO3_CONF_VARS']['FE']['contentRenderingTemplates']
...
+.. code-block:: yaml
+   :caption: EXT:my_extension/Configuration/Services.yaml
 
-   Array to register ContentObjects (cObjects) like :typoscript:`TEXT` or
-   :typoscript:`HMENU` within :file:`ext_localconf.php`,
-   see :file:`EXT:frontend/ext_localconf.php`
+    MyCompany\MyPackage\ContentObject\CustomContentObject:
+        tags:
+            - name: frontend.contentobject
+              identifier: 'MY_OBJ'
```
Comment (mbrodala) [inline Documentation/ApiOverview/GlobalValues/Typo3ConfVars/FE.rst:784]:
```suggestion
   services:
     # ...
     MyCompany\MyPackage\ContentObject\CustomContentObject:
       tags:
         - name: frontend.contentobject
           identifier: 'MY_OBJ'
```

**Review by mbrodala — CHANGES_REQUESTED** [review]

File: Documentation/ApiOverview/GlobalValues/Typo3ConfVars/FE.rst (hunk 1 above)
Comment (linawolf) [reply Documentation/ApiOverview/GlobalValues/Typo3ConfVars/FE.rst:773]: you can still use it if you want to make your extension compatible with both v11 and 12. But version 12 will not interpret it. Therefore I think my sentence is more specific

Comment (mbrodala) [reply Documentation/ApiOverview/GlobalValues/Typo3ConfVars/FE.rst:773]: If that was the idea, then we should explicitly mention this.

And my suggestion was only wording, it didn't have a different meaning. ;-)

Comment (linawolf) [reply Documentation/ApiOverview/GlobalValues/Typo3ConfVars/FE.rst]: @mbrodala  Ok, I reformulated the sentence once more, better like that?

Comment (linawolf) [reply Documentation/ApiOverview/GlobalValues/Typo3ConfVars/FE.rst]: @mbrodala do my changes resolve this comment?

**Pull request merged**